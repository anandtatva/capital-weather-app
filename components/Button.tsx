import { TouchableOpacity, Text, StyleSheet, ViewStyle, TouchableOpacityProps, TextStyle } from "react-native";
import { colors } from "../utils";


interface ButtonProps extends TouchableOpacityProps {
    onButtonPress: () => void;
    name: string;
    rootStyle?: ViewStyle;
    testId?: string;
    btnTextStyle?: TextStyle
}

function Button(props: ButtonProps) {
    const { name, btnTextStyle = {}, onButtonPress, testId, rootStyle = {}, disabled } = props;
    return (
        <TouchableOpacity
            testID={testId}
            onPress={onButtonPress}
            style={[styles.container, rootStyle, disabled && styles.disabled]}
            disabled={disabled}
        >
            <Text style={[styles.name, btnTextStyle]}>{name}</Text>
        </TouchableOpacity>
    )

}

const styles = StyleSheet.create({
    container: {
        backgroundColor: colors.primary,
        paddingVertical: 10,
        marginVertical: 20
    },
    name: {
        color: colors.secondary,
        textAlign: "center"
    },
    disabled: {
        opacity: 0.8
    }
})
export { ButtonProps }

export default Button;