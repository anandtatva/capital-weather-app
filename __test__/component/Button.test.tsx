import { render, waitFor, cleanup, fireEvent, act } from "@testing-library/react-native";
import Button, { ButtonProps } from "../../components/Button";

describe("Screen: Capital Weather", () => {

    it("renders correctly", () => {
        const props: ButtonProps = {
            onButtonPress: jest.fn(),
            name: "submit"
        }
        const tree = render(<Button {...props} />).toJSON()
        expect(tree).toMatchSnapshot();
    })

    it("Button should be call", () => {
        const props: ButtonProps = {
            onButtonPress: jest.fn(),
            testId: "Submit-Button",
            name: "submit"
        }
        const { getByTestId } = render(
            <Button {...props} />
        );
        const submitButton = getByTestId("Submit-Button")
        fireEvent.press(submitButton);
        expect(props.onButtonPress).toHaveBeenCalled();
    })
})