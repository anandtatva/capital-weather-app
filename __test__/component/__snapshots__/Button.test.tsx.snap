// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Screen: Capital Weather renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#000",
      "marginVertical": 20,
      "opacity": 1,
      "paddingVertical": 10,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#FFF",
          "textAlign": "center",
        },
        Object {},
      ]
    }
  >
    submit
  </Text>
</View>
`;
